dotnet/src/Microsoft.Agents.AI.Mem0/Mem0Client.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Agents.AI.Mem0;
+
+/// <summary>
+/// Client for the Mem0 memory service.
+/// </summary>
+internal sealed class Mem0Client
+{
+    private static readonly Uri s_searchUri = new("/v1/memories/search/", UriKind.Relative);
+    private static readonly Uri s_createMemoryUri = new("/v1/memories/", UriKind.Relative);
+
+    private readonly HttpClient _httpClient;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Mem0Client"/> class.
+    /// </summary>
+    /// <param name="httpClient">Configured <see cref="HttpClient"/> pointing at the Mem0 service (base address + auth headers).</param>
+    public Mem0Client(HttpClient httpClient)
+    {
+        this._httpClient = Throw.IfNull(httpClient);
+    }
+
+    /// <summary>
+    /// Searches for memories related to an input query.
+    /// </summary>
+    /// <param name="applicationId">Optional application scope.</param>
+    /// <param name="agentId">Optional agent scope.</param>
+    /// <param name="threadId">Optional thread scope.</param>
+    /// <param name="userId">Optional user scope.</param>
+    /// <param name="inputText">Query text.</param>
+    /// <param name="cancellationToken">Cancellation token.</param>
+    /// <returns>Enumerable of memory strings.</returns>
+    public async Task<IEnumerable<string>> SearchAsync(string? applicationId, string? agentId, string? threadId, string? userId, string? inputText, CancellationToken cancellationToken)
+    {
+        if (string.IsNullOrWhiteSpace(applicationId)
+            && string.IsNullOrWhiteSpace(agentId)
+            && string.IsNullOrWhiteSpace(threadId)
+            && string.IsNullOrWhiteSpace(userId))
+        {
+            throw new ArgumentException("At least one of applicationId, agentId, threadId, or userId must be provided.");
+        }
+
+        var searchRequest = new SearchRequest
+        {
+            AppId = applicationId,
+            AgentId = agentId,
+            RunId = threadId,
+            UserId = userId,
+            Query = inputText ?? string.Empty
+        };
+
+        using var content = new StringContent(JsonSerializer.Serialize(searchRequest, Mem0SourceGenerationContext.Default.SearchRequest), Encoding.UTF8, "application/json");
+        using var responseMessage = await this._httpClient.PostAsync(s_searchUri, content, cancellationToken).ConfigureAwait(false);
+        responseMessage.EnsureSuccessStatusCode();
+
+#pragma warning disable CA2016 // ReadAsStringAsync has no overload with token in current target; token already applied to request
+        var response = await responseMessage.Content.ReadAsStringAsync().ConfigureAwait(false);
```

Comment:
Suggest passing the token in the `ReadAsString` if `netstandard2.0` doesn't support, suggest assing the `#if` and remove the pragma.

```suggestion
        var response = await responseMessage.Content.ReadAsStringAsync(cancellationToken).ConfigureAwait(false);
```